source/ast/symbols/ClassSymbols.cpp

```diff
@@ -540,22 +540,23 @@ const Expression* ClassType::getBaseConstructorCall() const {
     return callExpr;
 }
 
-// Recursively finds interface classes that are implemented and adds them
+// Finds interface classes that are implemented and adds them
 // to the vector, if they haven't been added already.
 static void findIfaces(const ClassType& type, SmallVectorBase<const Type*>& ifaces,
-                       SmallSet<const Symbol*, 4>& visited) {
+                       SmallSet<const Symbol*, 4>& visited, bool recurse) {
     if (type.isInterface) {
         if (visited.emplace(&type).second)
             ifaces.push_back(&type);
     }
 
-    for (auto iface : type.getImplementedInterfaces()) {
+    for (auto iface :
```

Comment:
This seems wrong; it's still recursive either way -- you're pulling in each implemented interface's first level of parent interfaces as well. You don't need to call findIfaces() at all for the direct list, just add the interfaces in the loop below as you encounter them.